apps/igloo-kit-cli/src/infrastructure/olap/clickhouse.rs

```diff
@@ -219,14 +219,39 @@ pub fn create_client(clickhouse_config: ClickhouseConfig) -> ConfiguredDBClient
 
 // Run an arbitrary clickhouse query
 pub async fn run_query(
-    query: QueryString,
+    query: &QueryString,
     configured_client: &ConfiguredDBClient,
 ) -> Result<(), clickhouse::error::Error> {
     debug!("Running query: {:?}", query);
     let client = &configured_client.client;
     client.query(query.as_str()).execute().await
 }
 
+pub async fn check_ready(
+    configured_client: &ConfiguredDBClient,
+) -> Result<(), clickhouse::error::Error> {
+    let dummy_query = "SELECT version()".to_owned();
+    crate::utilities::retry::retry(
+        || run_query(&dummy_query, configured_client),
+        |i, e| {
+            i < 5
+                && match e {
+                    clickhouse::error::Error::Network(v) => {
+                        let err_string = v.to_string();
+                        err_string.contains("connection closed before message completed")
+                            || err_string.contains("connection error: Connection reset by peer")
+                            || err_string
+                                .contains("operation was canceled: connection was not ready")
+                            || err_string.contains("channel closed")
```

Comment:
`v` is `Box<dyn StdError + Send + Sync>`

Clickhouse is using an older version of hyper, so trying to downcast `v` into `hyper::Error` returns `None`. So I resorted to using string matching.